packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,297 @@
+    private get DATA_MAP(): AuthDataMap {
+        return {
+            access_token: {
+                storage: this.localStorage,
+                key: 'cc-ax',
+            },
+            customer_id: {
+                storage: this.localStorage,
+                key: 'customer_id',
+            },
+            usid: {
+                storage: this.localStorage,
+                key: 'usid',
+            },
+            enc_user_id: {
+                storage: this.localStorage,
+                key: 'enc_user_id',
+            },
+            expires_in: {
+                storage: this.localStorage,
+                key: 'expires_in',
+            },
+            id_token: {
+                storage: this.localStorage,
+                key: 'id_token',
+            },
+            idp_access_token: {
+                storage: this.localStorage,
+                key: 'idp_access_token',
+            },
+            token_type: {
+                storage: this.localStorage,
+                key: 'token_type',
+            },
+            refresh_token_guest: {
+                storage: this.cookieStorage,
+                key: 'cc-nx-g',
+                callback: () => {
+                    this.cookieStorage.delete('cc-nx')
+                },
+            },
+            refresh_token_registered: {
+                storage: this.cookieStorage,
+                key: 'cc-nx',
+                callback: () => {
+                    this.cookieStorage.delete('cc-nx-g')
+                },
+            },
+        }
+    }
+
+    private get data(): ShopperLoginTypes.TokenResponse {
+        return {
+            access_token: this.get('access_token'),
+            customer_id: this.get('customer_id'),
+            enc_user_id: this.get('enc_user_id'),
+            expires_in: parseInt(this.get('expires_in')),
+            id_token: this.get('id_token'),
+            idp_access_token: this.get('idp_access_token'),
+            refresh_token: this.get('refresh_token_registered') || this.get('refresh_token_guest'),
+            token_type: this.get('token_type'),
+            usid: this.get('usid'),
+        }
+    }
+
+    private isTokenExpired(token: string) {
+        if (!token) {
+            return true
+        }
+        const {exp, iat} = jwtDecode<JWTHeaders>(token.replace('Bearer ', ''))
+        const validTimeSeconds = exp - iat - 60
+        const tokenAgeSeconds = Date.now() / 1000 - iat
+        if (validTimeSeconds > tokenAgeSeconds) {
+            return false
+        }
+
+        return true
+    }
+
+    /**
+     * This function follows the public client auth flow to get
+     * access token for the user. Following the flow:
+     * 1. If we have valid access token - use it
+     * 2. If we have valid refresh token - refresh token flow
+     * 3. PKCE flow
+     *
+     * Only call this from the "ready" function, so "ready" can manage the pending state.
+     *
+     * @internal
+     */
+    private async init() {
```

Comment:
In my experiences "init" functions don't return a value, and if they do they typically return "this". Do we need to return anything here?

Reply:
+1 on `init` not returning anything.

Reply:
✅